packages/xstate-wallet/src/workflows/application.ts

```diff
@@ -0,0 +1,94 @@
+import {MachineConfig, Machine, StateMachine, send} from 'xstate';
+import {FINAL, MachineFactory} from '@statechannels/wallet-protocols';
+import {
+  CreateChannelEvent,
+  OpenChannelEvent
+} from '@statechannels/wallet-protocols/src/protocols/wallet/protocol';
+import {CreateChannel, JoinChannel} from '@statechannels/wallet-protocols/lib/src/protocols';
+import {IStore} from '@statechannels/wallet-protocols/src/store';
+import {State, getChannelId} from '@statechannels/nitro-protocol';
+import {SendStates} from '@statechannels/wallet-protocols/src/wire-protocol';
+
+// Events
+type OpenEvent = CreateChannelEvent | OpenChannelEvent;
+
+interface PlayerStateUpdate {
+  type: 'PLAYER_STATE_UPDATE';
+  state: State;
+}
+
+export type ApplicationWorkflowEvent = PlayerStateUpdate | SendStates | OpenEvent;
+
+// Context
+interface ApplicationContext {
+  channelId: string;
+}
+// States
+const initializing = {on: {CREATE_CHANNEL: 'funding', OPEN_CHANNEL: 'funding'}};
+
+const funding = {
+  invoke: {
+    id: 'openMachine',
+    src: 'invokeOpeningMachine',
+    data: (context, event) => event,
+    onDone: 'running',
+    autoForward: true
+  },
+  on: {SendStates: {actions: ['updateStore', 'sendChannelUpdated']}}
+};
+const running = {
+  on: {
```

Comment:
Are there times when one of these  events would arrive before entering the running state? How should they be handled then?

Reply:
This shouldn't ever happen but we could dump into an error state.